lib/model/weather/weather.dart

```diff
@@ -1,37 +1,22 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'package:flutter_training/common/result.dart';
+import 'package:flutter_training/model/weather/weather_info.dart';
+import 'package:flutter_training/model/weather/weather_request.dart';
 import 'package:yumemi_weather/yumemi_weather.dart';
 
-enum WeatherCondition {
-  sunny,
-  cloudy,
-  rainy,
-}
-
-extension _WeatherConditionExt on Iterable<WeatherCondition> {
-  WeatherCondition? byNameOrNull(String name) {
-    for (final element in this) {
-      if (element.name == name) {
-        return element;
-      }
-    }
-    return null;
-  }
-}
-
 class Weather {
   const Weather(this._weatherClient);
 
   final YumemiWeather _weatherClient;
 
-  Result<WeatherCondition?, String> fetchWeather() {
+  Result<WeatherInfo?, String> fetchWeather(WeatherRequest request) {
     try {
-      final condition = _weatherClient.fetchThrowsWeather('Tokyo');
-      final weatherCondition = WeatherCondition.values.byNameOrNull(condition);
-      if (weatherCondition == null) {
-        return const Result.failure('不明な天気予報です。');
-      }
-      return Result.success(weatherCondition);
+      final json = jsonEncode(request);
+      final weatherJsonData = _weatherClient.fetchWeather(json);
+      final weatherData = jsonDecode(weatherJsonData) as Map<String, dynamic>;
+      final response = WeatherInfo.fromJson(weatherData);
```

Comment:
[nits]
`weatherInfo` のほうが適切かなと思いました（修正漏れ？）

```suggestion
      final weatherInfo = WeatherInfo.fromJson(weatherData);
```

Reply:
おっと...修正漏れでした...。
ご指摘ありがとうございます！
こちらで修正しました。！
e40091359a087e16a82f8b59524343804cc5ba92